core/jreleaser-model-impl/src/main/java/org/jreleaser/model/spi/deploy/maven/Deployable.java

```diff
@@ -107,6 +107,11 @@ public Deployable(boolean snapshot, String stagingRepository, String path, Strin
             String v = version;
             int l = version.length();
             if (snapshot) {
+                // Maven requires snapshot versions to contain the literal "-SNAPSHOT" suffix
+                if (!version.contains("SNAPSHOT")) {
+                    throw new IllegalArgumentException(
+                        "Snapshot version '" + version + "' must contain '-SNAPSHOT' suffix for Maven deployment.");
```

Comment:
Please use message resources for this text. There should be other usages in this file that may be used as examplw

Reply:
> Please use message resources for this text. There should be other usages in this file that may be used as examplw

Done :+1: